### PR TITLE
Remove DDoS vulnerability with the connection limit

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
The commits explain everything.

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
Rebased on v0.x.x.

File: source/agora/common/Set.d [hunk 1]
```diff
@@ -83,8 +83,24 @@ private void fillFrom ( T ) ( ref T[] buffer, Set!T input )
 
 /// Pick random 'count' elements from the input set,
 /// and return an array of the elements (not very memory-efficient)
```
Comment (Geod24) [inline source/agora/common/Set.d]: What about this ?

Comment (AndrejMitrovic) [reply source/agora/common/Set.d]: I knew you were gonna catch this. :P

Yeah I'll add the note.

**Comment by AndrejMitrovic** [comment]
Fixed up the ddoc.

**Review by Geod24 — APPROVED** [review]
Looks much better now!

**Pull request merged**